app/utils/phase1_multimodal_extraction_util.py

```
@@ -0,0 +1,1868 @@
+    return overlap / min(aw, bw)
+
+
+def match_nearest_caption(
+    image_bbox: dict[str, float],
+    text_blocks: list[dict[str, Any]],
+) -> tuple[str, str | None, float | None]:
+    if not text_blocks:
+        return "", None, None
+
+    candidates: list[tuple[float, dict[str, Any], str]] = []
+    ix0, iy0, ix1, iy1 = image_bbox["x0"], image_bbox["y0"], image_bbox["x1"], image_bbox["y1"]
```

Ruff (0.15.2)

[error] Local variable `ix0` is assigned to but never used

Remove assignment to unused variable `ix0`

(F841)